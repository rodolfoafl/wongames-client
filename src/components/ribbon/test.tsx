import { render, screen } from '@testing-library/react'

import Ribbon from '.'

describe('<Ribbon />', () => {
  it('should render the children', () => {
    const { container } = render(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with the primary color', () => {
    render(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveClass('bg-primary')
  })

  it('should render with the secondary color', () => {
    render(<Ribbon color="secondary">Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveClass('bg-secondary')
  })

  it('should render with the default size', () => {
    render(<Ribbon>Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveClass('h-9 text-sm')
  })

  it('should render with the small size', () => {
    render(<Ribbon size="small">Best Seller</Ribbon>)

    expect(screen.getByText(/Best Seller/i)).toHaveClass('h-[1.625rem] text-xs')
  })
})
