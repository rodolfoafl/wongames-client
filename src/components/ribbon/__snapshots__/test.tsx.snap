// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> should render the children 1`] = `
<div
  class="absolute top-4 flex items-center font-bold text-white right-0 before:hidden before:brightness-75 lg:before:block before:absolute before:right-0 before:border before:border-b-[0.625rem] before:border-l-0 before:border-r-transparent before:border-b-transparent bg-primary before:border-l-primary before:border-t-primary h-9 px-6 py-0 text-sm before:top-9 before:border-t-[0.625rem] before:border-r-[1.25rem] lg:-right-5"
>
  Best Seller
</div>
`;
